document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/transformer/FieldToPgColumnTransformer.java

```diff
@@ -0,0 +1,41 @@
+package org.hypertrace.core.documentstore.postgres.query.v1.transformer;
+
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.core.documentstore.postgres.query.v1.PostgresQueryParser;
+import org.hypertrace.core.documentstore.postgres.utils.PostgresUtils;
+
+public class FieldToPgColumnTransformer {
+  private static final String DOT = ".";
+
+  private PostgresQueryParser postgresQueryParser;
+
+  public FieldToPgColumnTransformer(PostgresQueryParser postgresQueryParser) {
+    this.postgresQueryParser = postgresQueryParser;
+  }
+
+  public FieldToPgColumn transform(String orgFieldName) {
+    Optional<String> parentField =
+        postgresQueryParser.getPgColumnNames().keySet().stream()
+            .filter(f -> orgFieldName.startsWith(f))
```

Comment:
1. Are we saying that the alias can only be a part of the field name?
2. Is it the reason for handling only small cases?

Reply:
No, even if I pass the aliases to SQL query, the return column name from Postgresql is in lowerCase as default.
e.g
`Select document->'item' as ITEM from myTest` will return me `item` as a column. So, the Document iterator has no knowledge of lower/upper case while parsing the result. Prior (Query API v0), we were not supporting selection, so there was no need. But, with the new Query interface, we need to support alias with either case. As this is independent of `UnnestExpression`,  looking at it as a separate PR. Working on it.